packages/wonder-blocks-button/components/button.js

```diff
@@ -1,52 +1,68 @@
 // @flow
 import React from "react";
 import {StyleSheet, css} from "aphrodite";
+import ButtonCore from "./button-core.js";
 
-import Color from "wonder-blocks-color";
-import {LabelLarge} from "wonder-blocks-typography";
+import type {SharedProps} from "../types.js";
 
-type Props = {
-    color?: "blue" | "green" | "gold" | "red",
+type Props = SharedProps & {
+    /**
+     * URL to navigate to.
+     *
+     * Note: Either href or onClick must be defined
+     */
+    href?: string,
+
+    /**
+     * Whether to use client-side navigation.
+     *
+     * If the URL passed to href is local to the client-side, e.g.
+     * /math/algebra/eval-exprs, then use ReactRouter to do a client side
+     * navigation by doing history.push(this.props.href) using
+     * ReactRouter's history object
+     */
+    clientSideNav?: boolean,
+
+    /**
+     * Function to call when button is clicked.
+     *
+     * This callback should be used for things like marking BigBingo
+     * conversions. It should NOT be used to redirect to a different URL or to
+     * prevent navigation via e.preventDefault(). The event passed to this
+     * handler will have its preventDefault() and stopPropagation() methods
+     * stubbed out.
+     *
+     * Note: onClick is optional if href is present, but must be defined if
+     * href is not
+     */
     onClick?: (e: SyntheticEvent<MouseEvent | TouchEvent>) => void,
 };
 
 export default class Button extends React.Component<Props> {
     render() {
         return (
-            <a
-                className={css(
-                    styles.button,
-                    styles[this.props.color || "blue"],
-                )}
-                onClick={this.props.onClick}
+            <div
+                onClick={this.props.onClick && this.props.onClick}
+                href={this.props.href && this.props.href}
             >
-                <LabelLarge>Hello, world!</LabelLarge>
-            </a>
+                <ButtonCore
+                    icon={this.props.icon ? this.props.icon : undefined}
```

Comment:
Hmm... I think you can simplify a lot of this by doing: `{...this.props}` and removing all of the props that you're just passing through.